agent:backhaul_manager:really parse tlvHigherLayerData

CoralMalachi opened the pull request:
Fix a bug that has been added in commit d491828.

The problem is addClass is performed not on the right CMDU, but on an
empty CMDU, which causes the payload length to be zero.

Therefore, perform the addClass method on the right CMDU.

Signed-off-by: Coral Malachi <coral.malachi@intel.com>

Commit:
agent:backhaul_manager:really parse tlvHigherLayer

Fix a bug that has been added in commit d491828

The problem is addClass is performed not on the right CMDU, but on an
empty CMDU, which causes the payload length to be zero.

Therefore, perform the addClass method on the right CMDU.

Signed-off-by: Coral Malachi <coral.malachi@intel.com>

Commit:
agent:backhaul_manager:print payload length

Change the `handle_1905_higher_layer_data_message` to print the payload
length instead of the TLV length.

Signed-off-by: Coral Malachi <coral.malachi@intel.com>

Commit:
agent:backhaul: unified variables declaration

Change the declaration of variables to be uniform, as well as the print
format.

Signed-off-by: Coral Malachi <coral.malachi@intel.com>

tomereli commented on agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp:
```diff
@@ -1647,16 +1647,16 @@ bool main_thread::handle_1905_higher_layer_data_message(ieee1905_1::CmduMessageR
 
-    auto protocol     = tlvHigherLayerData->protocol();
-    const auto length = tlvHigherLayerData->length();
+    const auto protocol       = tlvHigherLayerData->protocol();
```
Is this related to this fix?

CoralMalachi replied:
I wanted it to be uniform, Do you want me to split it to another commit?

tomereli replied:
yes please

CoralMalachi replied:
Ok :)

arnout reviewed (changes requested):
Could you also extend the test so it catches this issue? I guess that matching the protocol and length in the received message should be enough. For that, it's probably also easier to put everything in one line.

Also, the commit 97e8d8e is not in master. In master, it's d491828 (due to rebase when merging, github is a bit stupid in that respect).

Finally, a commit summary that just says "Fix a bug that has been added" is not very useful. Better would be: `agent: backhaul_manager: really parse tlvHigherLayerData`, followed by the rest of you message (i.e. "Fix a bug ...".

Commit:
tools:docker: check for matching values

Check that the protocol name and payload length in the received message
are matching to the values that were declared in the MCUT's command.

Signed-off-by: Coral Malachi <coral.malachi@intel.com>
